lib/membrane/rtp/demuxer.ex

```diff
@@ -309,55 +321,87 @@ defmodule Membrane.RTP.Demuxer do
     {[], state}
   end
 
+  @spec get_output_actions(JitterBuffer.State.t(), stream_phase()) ::
+          {[Membrane.Element.Action.t()], JitterBuffer.State.t()}
+  defp get_output_actions(jitter_buffer_state, stream_phase) do
+    case stream_phase do
+      :bound_with_pad ->
+        JitterBuffer.get_output_actions(jitter_buffer_state)
+
+      _invalid_phase ->
+        {[], jitter_buffer_state}
+    end
+  end
+
   @spec initialize_new_stream_state(
-          matching_pad :: Pad.ref() | nil,
+          Pad.ref() | nil,
           ExRTP.Packet.t(),
           CallbackContext.t(),
           State.t()
         ) :: {[Membrane.Element.Action.t()], State.t()}
-  defp initialize_new_stream_state(nil, packet, _ctx, state) do
+  defp initialize_new_stream_state(pad_waiting_for_stream, packet, ctx, state) do
     jitter_buffer_state = JitterBuffer.new(packet)
 
     stream_state = %State.StreamState{
-      phase: :waiting_for_matching_pad,
-      pad: nil,
-      pad_match_timer:
-        Process.send_after(
-          self(),
-          {:pad_match_timeout, packet.ssrc},
-          Membrane.Time.as_milliseconds(state.not_linked_pad_handling.timeout, :round)
-        ),
       payload_type: packet.payload_type,
       jitter_buffer_state: jitter_buffer_state
     }
 
-    state = put_in(state.stream_states[packet.ssrc], stream_state)
+    if pad_waiting_for_stream != nil do
+      {actions, stream_state} =
+        bind_pad_with_stream(
+          pad_waiting_for_stream,
+          ctx.pads[pad_waiting_for_stream].options,
+          stream_state,
+          state.payload_type_mapping
+        )
 
-    {[notify_parent: create_new_stream_notification(packet)], state}
+      {_stream_id, state} = pop_in(state.pads_waiting_for_stream[pad_waiting_for_stream])
+      state = put_in(state.stream_states[packet.ssrc], stream_state)
+      {actions, state}
+    else
+      stream_state = %State.StreamState{
```

Comment:
[NIT] It would also be fine this way you Static-Fanatic ;)
```suggestion
      stream_state = %{
```